File: back-end/src/main/java/co/edu/unihumboldt/parking/mapping/dtos/FeeDto.java
package co.edu.unihumboldt.parking.mapping.dtos;

import co.edu.unihumboldt.parking.domain.entities.ParkingFacility;
import co.edu.unihumboldt.parking.domain.entities.VehicleType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
/**
 * Clase {@code FeeDto} que representa un objeto de transferencia de datos para una tarifa.
 * Contiene atributos que describen el nombre de la tarifa, su precio, estado (activo/inactivo),
 * el parque de estacionamiento asociado y el tipo de vehículo relacionado.
 * Utiliza Lombok para generar constructores, métodos de acceso y un constructor de estilo "builder".
 */

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class FeeDto {
    private int id;
    private String name;
    private int price;
    private boolean status;
    private ParkingFacility parkingFacility;
    private VehicleType vehicleType;
}

File: front-end/app-vue/src/models/FeeDto.ts
import type {VehicleTypeDto} from "@/models/VehicleTypeDto";
import type {ParkingFacility} from "@/models/ParkingFacility";

export interface FeeDto {
    id: number;
    name: string;
    price: number;
    status: boolean;
    parkingFacility: { id: number };
    VehicleTypeDto: {id: number};
}